database/mongoSeed.js

```diff
@@ -0,0 +1,100 @@
+let db = require('./mongo.js');
+let mongoose = require('mongoose');
+const randomNumberUpTo = (limit) => Math.floor(Math.random() * limit);
+
+var listingSchema = new mongoose.Schema({
+  id: Number,
+  price: Number,
+  maxGuests: Number,
+  minStay: Number,
+  stars: mongoose.Decimal128,
+  numRatings: Number,
+  bookedDates: [String]
+})
+
+var Listing = mongoose.model('Listing', listingSchema);
+
+// create listings and insert in batches of 100
+const seedListings = async () => {
+  let t1 = Date.now();
+  for (let batch = 0; batch < 100000; batch++) {
+    let listingBatch = [];
+    for (let listing = 0; listing < 100; listing++) {
+      let id = (batch * 100) + (listing + 1);
+      let price = 50 + randomNumberUpTo(400);
+      let maxGuests = 1 + randomNumberUpTo(6);
+      let minStay = 1 + randomNumberUpTo(2);
+      let stars = (1 + (Math.random() * 4)).toFixed(2);
+      let numRatings = randomNumberUpTo(110);
+      let bookedDates = await generateBookedDates(minStay);
+      let listingProps = {
+        id,
+        price,
+        maxGuests,
+        minStay,
+        stars,
+        numRatings,
+        bookedDates
+      }
+      let newListing = new Listing(listingProps);
+      listingBatch.push(newListing);
+    }
+    await Listing.insertMany(listingBatch)
+    console.log(`${100*batch + 100} listings inserted`);
+  }
+  let timeElapsed = Date.now() - t1;
+  console.log(`timeElapsed: ${timeElapsed}`);
+}
+
+seedListings();
+
+async function generateBookedDates(minStay) {
+
+  // create objects to map months to corresponding number of days in month and to year
+  let months = [11,12,1,2];
+  let daysInMonth = {
+    11: 30,
+    12: 31,
+    1: 31,
+    2: 28
+  }
```

Comment:
This looks fine -- if you want to refactor in the future, I found that Moment.js is very easy to use/quick to integrate and makes it easy to generate dates within a range. It would probably allow you to skip over these mappings and generalize to any time interval you'd like.

Not a suggestion for this PR, just for refactoring in general if you'd like to try a new library.